daft/dataframe/dataframe.py

```diff
@@ -898,7 +901,11 @@ def count(self, *cols: ColumnInputType) -> "DataFrame":
         Returns:
             DataFrame: Globally aggregated count. Should be a single row.
         """
-        assert len(cols) > 0, "no columns were passed in"
+        if len(cols) == 0:
+            logger.warning(
+                "No columns specified; performing count on all columns. Specify columns using df.count('col1', 'col2', ...) or use df.count_rows() for row counts."
+            )
+            cols = tuple(self.columns)
```

Comment:
Hi @samster25 ,

Could you take a look at this and lmk if this aligns with what you expected? Once it's all good with you I'll modify the rest of the agg methods.

Reply:
This looks good to me! Though I wonder if we should corner-case the `.count()` aggregation method to just forward the call to `.count_rows()` instead 😛 . I believe that in PostgreSQL for example, `COUNT(*)` is actually a row count of the entire result set vs `COUNT(c)` which does a null-aware count of a column.

These semantics sound good for all the other aggregation-type methods though!

Reply:
Makes sense, though `.count_rows()` returns an int whereas `.count()` should return a dataframe, so instead of simply forwarding the call we could do `return DataFrame(self._builder.count())`? which would result in a dataframe that looks like:
```
+--------+
| count  |
| UInt64 |
+--------+
| 3      |
+--------+
```

Reply:
Oh yes good point. The current semantics of this PR makes perfect sense then 👍

Reply:
sweet, thanks! just made changes for the rest of the aggregation methods in latest commit

Reply:
Also, some of the integration tests are failing with `Error: Credentials could not be loaded, please check your action inputs: Could not load credentials from any providers`, [example](https://github.com/Eventual-Inc/Daft/actions/runs/6829508870/job/18575937012?pr=1587#step:8:16). Do you know what's the cause 😅 ?

the release drafter label test too: `[Error: Resource not accessible by integration](https://github.com/Eventual-Inc/Daft/actions/runs/6829508921/job/18575800562?pr=1587#step:2:25)`

Reply:
Oh yes that's security related because these CI tests are actually running on your fork, not on the main repo 😛 

We need to figure out a (secure) way to run our integration tests on incoming PRs, or maybe a better policy here around accepting external contributions.